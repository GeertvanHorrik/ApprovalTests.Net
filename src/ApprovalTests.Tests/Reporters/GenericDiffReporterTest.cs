using ApprovalTests.Reporters;
using ApprovalUtilities.Utilities;
using NUnit.Framework;
using System.IO;
using ApprovalTests.Reporters.Windows;

namespace ApprovalTests.Tests.Reporters
{
    [TestFixture]
    public class GenericDiffReporterTest
    {
        [Test]
        public void TestGetActualProgramFileEchos()
        {
            var NoneExistingFile = @"C:\ThisDirectoryShouldNotExist\ThisFileShouldNotExist.exe";
            Assert.AreEqual(NoneExistingFile, GenericDiffReporter.GetActualProgramFile(NoneExistingFile));
        }


        [Test]
        public void TestProgramsExist()
        {
            Assert.IsFalse(new GenericDiffReporter("this_should_never_exist", "").IsWorkingInThisEnvironment("any.txt"));
        }

        [Test]
        [UseReporter(typeof(TortoiseDiffReporter))]
        public void TestEnsureFileExist()
        {
            var imageFile = PathUtilities.GetAdjacentFile("TestImage.png");
            if (File.Exists(imageFile))
            {
                File.Delete(imageFile);
            }

            GenericDiffReporter.EnsureFileExists(imageFile);
            Approvals.VerifyFile(imageFile);
        }
    }
}