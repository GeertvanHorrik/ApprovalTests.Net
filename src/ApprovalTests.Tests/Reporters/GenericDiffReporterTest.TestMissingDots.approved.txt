System.ArgumentException: The following extensions don't start with dots: [txt, asp]